packages/office-ui-fabric-react/src/components/DetailsList/DetailsList.tsx

```diff
@@ -577,7 +577,21 @@ export class DetailsList extends BaseComponent<IDetailsListProps, IDetailsListSt
         this._rememberCalculatedWidth(column, column.calculatedWidth!);
       });
     } else {
-      adjustedColumns = this._getJustifiedColumns(newColumns, viewportWidth, newProps);
+      if (resizingColumnIndex !== undefined) {
+        adjustedColumns = this._getJustifiedColumnsAfterResize(newColumns, viewportWidth, newProps, resizingColumnIndex);
+      } else {
+        adjustedColumns = this._getJustifiedColumns(newColumns, viewportWidth, newProps, 0);
+      }
+
+      // Last column is not resizable if there's no column behind.
+      // Otherwise, it can be resized to bring back any dropped column.
+      if (adjustedColumns.length && adjustedColumns.length === newColumns.length) {
+        adjustedColumns[adjustedColumns.length - 1].isResizable = false;
+      }
```

Comment:
What if I want to make the last column wider than the screen?

Reply:
This is not new, justified layout has always disable resizing of last column. I've only relaxed the rule.
I see 2 scenarios:
- I don't want horizontal scroll bar and I rely on dropping columns. Then it makes sense that last column is not resizable.
- I want justified layout, but I'm ready to display a scroll bar on overflow (like when columns are not collapsable, which means they won't drop). Then I see value on letting it be resizable.

So I'm going ahead and deleting this entirely, actually it can be implemented in the columns collection if I don't want last column to be resizable.

Reply:
Sorry, it's not that simple. Last column is constrained by available width because we're on justified mode. So we would need more changes than just removing that logic, and also need to consider edge cases between dropping columns and resizing to overflow.
As I said, I'm not opposed to it, but I would prefer working on it in a separate work. This PR is relaxing the constraint, a next one can drop it entirely once we take care of those details.

Reply:
But not being able to make the columns wider than the screen, wasn't that the whole issue why we had to make that hotfix?

Reply:
No, the hotfix was about `maxWidth`, the viewport has never limited.
In the Fabric Documents example, you would need to remove `isCollapsible` from columns to make them wider than viewport (`overflow:auto` may also help). Otherwise, you're testing the "I don't want horizontal scrollbar" scenario.

That said, I've received more feedback aligned with your question, so I'm for allowing last column resizing to overflow. It should be worked out separately though.

Reply:
Working on your feedback, I've found a way to implement it that satisfies me and doesn't add effort to this PR. Now last column is always resizable (unless it's disabled in columns collection), and user can always make any column wider, even if that means overflowing the justified layout (so a horizontal scrollbar should appear in that case, but that's the host responsibility).